cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/templates/etl/batch/BatchDriver.java

```diff
@@ -0,0 +1,206 @@
+import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Lists;
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.Mapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * MapReduce driver for Batch ETL Adapters.
+ */
+public class BatchDriver extends AbstractMapReduce {
+  private static final Logger LOG = LoggerFactory.getLogger(BatchDriver.class);
+  private static final Gson GSON = new Gson();
+  private static final JsonParser JSON_PARSER = new JsonParser();
+
+  private BatchSource source;
+  private BatchSink sink;
+
+  @Override
+  public void configure() {
+    setName("BatchDriver");
+    setDescription("MapReduce driver for Batch ETL Adapters");
+  }
+
+  @Override
+  public void beforeSubmit(MapReduceContext context) throws Exception {
+    Job job = context.getHadoopJob();
+
+    Preconditions.checkArgument(context.getRuntimeArguments().containsKey(Constants.Source.CLASS_NAME));
+    Preconditions.checkArgument(context.getRuntimeArguments().containsKey(Constants.Sink.CLASS_NAME));
+    Preconditions.checkArgument(context.getRuntimeArguments().containsKey(Constants.Transform.TRANSFORM_CLASS_LIST));
+
+    String sourceClass = context.getRuntimeArguments().get(Constants.Source.CLASS_NAME);
+    String sinkClass = context.getRuntimeArguments().get(Constants.Sink.CLASS_NAME);
+
+    source = (BatchSource) Class.forName(sourceClass).newInstance();
+    sink = (BatchSink) Class.forName(sinkClass).newInstance();
+
+    prepareSourceJob(context);
+    prepareSinkJob(context);
+    prepareTransform(context);
+
+    job.setMapperClass(MapperDriver.class);
+    job.setNumReduceTasks(0);
+  }
+
+  private void prepareSourceJob(MapReduceContext context) {
+    DefaultStageConfigurer sourceConfigurer = new DefaultStageConfigurer(source.getClass());
+    source.configure(sourceConfigurer);
+    BatchSourceContext sourceContext = new MapReduceSourceContext(context, sourceConfigurer.createSpecification());
+    source.prepareJob(sourceContext);
+  }
+
+  private void prepareSinkJob(MapReduceContext context) {
+    DefaultStageConfigurer sinkConfigurer = new DefaultStageConfigurer(sink.getClass());
+    sink.configure(sinkConfigurer);
+    BatchSinkContext sinkContext = new MapReduceSinkContext(context, sinkConfigurer.createSpecification());
+    sink.prepareJob(sinkContext);
+  }
+
+  private void prepareTransform(MapReduceContext context) {
+    Job job = context.getHadoopJob();
+    JsonObject config = JSON_PARSER.parse(context.getRuntimeArguments().get(Constants.CONFIG_KEY)).getAsJsonObject();
```

Comment:
we want to avoid working with json directly.  This should be a java object with source, sink, and a list of transforms as its fields. We can then use gson to deserialize it into that object